qiskit_nature/circuit/library/ansatzes/ucc.py

```diff
@@ -354,6 +358,43 @@ def _get_excitation_generators(self) -> List[Callable]:
 
         return generators
 
+    def _check_excitation_list(self, excitations: Sequence) -> None:
+        """Checks the format of the given excitation operators.
+
+        The following conditions are checked:
+        - the list of excitations consists of pairs of tuples
+        - each pair of excitation indices has the same length
+        - the indices within each excitation pair are unique
+
+        Args:
+            excitations: the list of excitations
+
+        Raises:
+            QiskitNatureError: if format of excitations is invalid
+        """
+        logger.debug("Checking excitation list...")
+
+        if any(len(exc) != 2 for exc in excitations):
+            raise QiskitNatureError(
```

Comment:
What might be helpful if this message gets raised is to have the excitation list that is in error so the user can compare. We normally try and have the failing value in an exception message. Maybe its too much to identify the failure within that list somehow, but maybe the list is long hence including it all may may rather a long message!

Reply:
I've updated the error message. The checks are now executed for each excitation individually. If one of the checks fails the excitation is included in the error message.